nucliadb_node/src/replication/replicator.rs

```diff
@@ -134,9 +134,14 @@ pub async fn replicate_shard(
     drop(file);
     drop(_gc_lock);
 
-    if generation_id.is_some() {
+    if let Some(gen_id) = generation_id {
```

Comment:
just a comment, on line 87 you are doing this:
```rust
generation_id = Some(resp.generation_id);
```
which makes this code kind of unnecessary, maybe there is some logic you are not taking into account? If not, why not doing:
```rust
shard.metadata.set_generation_id(resp.generation_id);
```
and remove the mutable `gen_id` var that only seems to add noise.

Reply:
oh, that is weird and needs to be cleaned up! Thanks :)

Reply:
oh, I was trying not to set the gen id in the while

Reply:
But you have it already in `resp`, you dont need to declare anything. Specially because I see its not used after the loop.

Reply:
the response is the scope of the while loop. Technically, the stream could be empty and we have no value

Reply:
I see